### PR TITLE
Added `run_bechnmark` function

### PR DESCRIPTION
**Pull request by vo-nil**
Added templated function `run_benchmark<group_type...>(lambda)`.

It initializes samples array of given template types and runs series of calls to lambda. Samples are traversed with strides so they do not sit in cache.

After the benchmark the statistics are printed to `std::cout`.

### CODE REVIEW

File: libs/bench/include/nil/crypto3/bench/benchmark.hpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    stddiv = sqrt(stddiv / durations.size() - mean * mean);
+
+    // https://support.numxl.com/hc/en-us/articles/115001223503-MdAPE-Median-Absolute-Percentage-Error
+    for(auto &dur : durations) {
+        dur = (dur - median) / dur;
+        if ( dur < 0 ) {
+            dur = -dur;
+        }
+    }
+    std::sort(durations.begin(), durations.end());
```
Comment (martun) [inline libs/bench/include/nil/crypto3/bench/benchmark.hpp]: Probably you don't need to sort the durations again after normalization.

File: libs/marshalling/zk/example/CMakeLists.txt [hunk 2]
```diff
@@ -15,15 +15,11 @@ macro(define_marshalling_example name)
     add_executable(marshalling_${name}_example ${name}.cpp)
 
     target_link_libraries(marshalling_${name}_example PRIVATE
-
                           crypto3::multiprecision
                           crypto3::algebra
                           crypto3::zk
                           ${CMAKE_WORKSPACE_NAME}::marshalling-multiprecision
                           ${CMAKE_WORKSPACE_NAME}::marshalling-algebra
-                          ${CMAKE_WORKSPACE_NAME}::crypto3_zk
```
Comment (martun) [inline libs/marshalling/zk/example/CMakeLists.txt:24]: Our examples don't compile anyway right? We can still  remove these, but that will not help right?

File: libs/bench/include/nil/crypto3/bench/benchmark.hpp (hunk 1 above)
Comment (vo-nil) [reply libs/bench/include/nil/crypto3/bench/benchmark.hpp]: no, we need. After APE is calculated, we need to get median. The first half is negative, the least values are in the middle and the last part is positive. and we get abs values. 
Consider this (sorted) array: 1 2 3 4 5. It has median = 3. When calculating MdAPE, we get this array:
`-2 -0.5 0 0.25 0.4`
We take absolute values and take median of that array. Yes, it needs to be sorted.

File: libs/marshalling/zk/example/CMakeLists.txt (hunk 2 above)
Comment (vo-nil) [reply libs/marshalling/zk/example/CMakeLists.txt:24]: At least cmake won't  complain if the project is built with `BUILD_EXAMPLES=TRUE`

**Review by martun — APPROVED** [review]

**Pull request merged**